### PR TITLE
Transaction status

### PR DESCRIPTION
**Pull request by PunkSage**

### CODE REVIEW

File: src/redux/actions/TransactionAction.js [hunk 1]
```diff
@@ -1,35 +1,38 @@
...
-      promise.then((tx) => { 
-        dispatch(statusAddPendingTransaction(tx, caption, url));
-        dispatch(watchPendingTransaction(tx, caption, url));
-      }).catch((reason) => {
-        dispatch(statusAddFailedTransaction("", caption, reason));
-      });
-  }  
-}
+    return function (dispatch) {
+        promise.then((address) => {
```
Comment (vanruch) [inline src/redux/actions/TransactionAction.js:11]: Maybe call it transactionHash?

**Review by vanruch — APPROVED** [review]

File: src/redux/actions/TransactionAction.js (hunk 1 above)
Comment (PunkSage) [reply src/redux/actions/TransactionAction.js:11]: OK, sounds good.

File: src/redux/actions/MarketAction.js [hunk 2]
```diff
@@ -41,133 +41,154 @@ const createMarketResponse = (marketContract) => {
...
         dispatch(requestNewMarket());
         await waitForAmbrosus();
-        var marketRepository = new Ambrosus.MarketRepository(Ambrosus.marketArtifacts);
+        let marketRepository = new Ambrosus.MarketRepository(Ambrosus.marketArtifacts);
         marketRepository.create(web3.eth.accounts[0], (transactionHash) => {
-            dispatch(statusAddPendingTransaction(transactionHash, "Creating contract", "ads"));
+            dispatch(statusAddPendingTransaction({
+                address: transactionHash,
+                caption: "Creating contract",
+                url: "http://www.google.pl"
```
Comment (marekkirejczyk) [inline src/redux/actions/MarketAction.js]: /market

File: src/redux/actions/MarketAction.js [hunk 3]
```diff
@@ -41,133 +41,154 @@ const createMarketResponse = (marketContract) => {
...
+            }));
             dispatch(createMarketResponse(transactionHash));
             dispatch(showModal("TransactionProgressModal", {title: "Creating market"}));
         }).then((myContract) => {
-            dispatch(statusAddSuccessTransaction(myContract.marketContract.transactionHash, "Creating contract", "ads"));
-            dispatch(createMarketSuccess(myContract.marketContract.address));
+            dispatch(statusAddSuccessTransaction({
+                address: myContract.marketContract.transactionHash,
+                caption: "Creating contract",
+                url: "http://www.google.pl"
```
Comment (marekkirejczyk) [inline src/redux/actions/MarketAction.js]: /market

File: src/redux/actions/MarketAction.js [hunk 4]
```diff
@@ -41,133 +41,154 @@ const createMarketResponse = (marketContract) => {
...
             dispatch(redirectToMarket(history));
         }).catch((err) => {
-            dispatch(statusAddFailedTransaction("", "Creating contract", err));
-            dispatch(showModal("ErrorModal", { reason: err }));
+            dispatch(statusAddFailedTransaction(
+                {
+                    address: myContract.marketContract.address,
+                    caption: "Creating contract",
+                    url: "http://www.google.pl"
+                }));
```
Comment (marekkirejczyk) [inline src/redux/actions/MarketAction.js]: /market

File: src/redux/reducers/TransactionsStatusReducer.js [hunk 5]
```diff
@@ -1,47 +1,50 @@
...
+                list: [...filtered, action.data],
+                stats: {
+                    approved: state.stats.approved,
+                    failed: state.stats.failed + 1,
+                    pending: state.stats.pending - 1,
+                }
             };
-
-        case 'STATUS_ADD_SUCCESS_TRANSACTION':
+            break;
```
Comment (marekkirejczyk) [inline src/redux/reducers/TransactionsStatusReducer.js:38]: Why break after return?

File: src/redux/actions/MarketAction.js [hunk 6]
```diff
@@ -41,154 +41,175 @@ const createMarketResponse = (marketContract) => {
...
         const requirementsRepository = new Ambrosus.RequirementsRepository();
         const requirements = await requirementsRepository.getAllFromMarket(market);
-        var names = await requirementsNames(requirements);
+        let names = await requirementsNames(requirements);
         dispatch(receiveAllRequirements(names));
     };
-}
+};
 
 async function requirementsNames(requirements) {
```
Comment (marekkirejczyk) [inline src/redux/actions/MarketAction.js:209]: @vanruch I think that function should be implemented in Contract

File: src/redux/actions/MarketAction.js [hunk 7]
```diff
@@ -41,154 +41,175 @@ const createMarketResponse = (marketContract) => {
...
+                url: "ads"
+            }));
+            dispatch(successCreateRequirement(requirements.contract.address));
             dispatch(hideModal());
             dispatch(redirectToMarket(history));
         }).catch((err) => {
-            console.error(err)
-            dispatch(statusAddFailedTransaction("", "Creating contract", err));
-            dispatch(showModal("ErrorModal", { reason: err }));
+            console.error(err);
```
Comment (marekkirejczyk) [inline src/redux/actions/MarketAction.js:168]: Now that we push error to UI, perheps we don't need to put it to console?

**Pull request merged**